Form/ChoosePaymentMethodType.php

```diff
@@ -92,15 +92,18 @@ protected function buildChoiceList(FormBuilderInterface $builder, array $options
             $label = 'form.label.'.$method;
 
             if (Legacy::formChoicesAsValues()) {
-                $options['choices'][$method] = $label;
-            } else {
                 $options['choices'][$label] = $method;
+                if (Legacy::needsChoicesAsValuesOption()) {
+                    $options['choices_as_values'] = true;
```

Comment:
I don't think this is correct. I don't recall the details of the `choices_as_values` option (and as you mention in a comment, this isn't easy to grasp) but this changes the behaviour. I'm pretty sure we don't want to set that to true, in order to maintain BC with projects using Symfony < 2.7.